.github/workflows/build.yml

```
@@ -0,0 +1,38 @@
+name: build
+
+on: [push]
```

```suggestion
on:
  pull_requests:
    paths-ignore:
      - '*.rst'
      - 'UNLICENCE'
      - 'Doxyfile.in'
      - 'shaders/**'
  push:
    paths-ignore:
      - '*.rst'
      - 'UNLICENCE'
      - 'Doxyfile.in'
      - 'shaders/**'
```
1. Change to also run the action when someone opens or updates a pull request. That way build issues can be resolved before merging.
2. Ignore changes to text documents and shaders, as neither are being processed during compilation

It’s possible I made some mistakes in specifying things, as I am still discovering how GitHub actions works. I am not sure if there is a better way to specify those paths-ignore without needing to duplicate them for both pull requests and push events.

---

1. I personally haven't found any use for the `pull_request` trigger. If you have the `push` trigger enabled, then every push will be built, even if it's in another branch and in a PR. So you won't be able to merge changes from a PR without them being built.
The times I've tried having `push` and `pull_request` on at the same time, it has just started two identical builds at the same time.
2. Makes sense.

---

1. From what I can find online, `push` and `pull_request` can both end up being called if one opens an internal PR (i.e. the branch to be merged is located in the same repository where the merge will take place), but it does not seem to occur for external PR, like yours where the branch is located in your fork. There was a [comment](https://stackoverflow.com/a/65946732) on StackOverflow on how to prevent double builds in the case of internal PRs.

    Shouldn’t GitHub Actions be already running for this PR, or will it only start testing PRs once this one has been merged?

---

Still does not seem to be triggering on this PR, despite adding the `pull_request`. I guess we will have to revisit this part after merging this PR.

---

I'm pretty sure it will only trigger builds on this repository once you've merged the PR because of security reasons. You don't want people to create malicious workflows and run them in your repository by creating a PR.

---

There are restrictions at the repository level: for example by default, one has to approve running workflows for first time contributors, to work against the security issues. Though I don’t get a prompt on this PR for approving the workflow.